kubectl-cloudflow/cmd/verify.go

```diff
@@ -0,0 +1,170 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-akka/configuration"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/domain"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/util"
+	"github.com/lightbend/cloudflow/kubectl-cloudflow/verify"
+	"github.com/spf13/cobra"
+)
+
+type verifyBlueprintCMD struct {
+	cmd *cobra.Command
+}
+
+func init() {
+	verifyBlueprintCMD := &verifyBlueprintCMD{}
+	verifyBlueprintCMD.cmd = &cobra.Command{
+		Use:     "verify",
+		Short:   "Verifies the blueprint of a Cloudflow Application.",
+		Example: "kubectl cloudflow verify <blueprint-file-name>",
+		Run:     verifyBlueprintCMD.verifyImpl,
+		Args:    validateVerifyCmdArgs,
+	}
+	rootCmd.AddCommand(verifyBlueprintCMD.cmd)
+}
+
+// this map is constructed entirely from blueprint
+func getImageRefsFromConfig(config *configuration.Config) map[string]domain.ImageReference {
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageKeyVals = make(map[string]domain.ImageReference)
+
+	for imageKey, imageRef := range images {
+		ref, err := verify.ParseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit(err.Error())
+		}
+		imageKeyVals[imageKey] = *ref
+	}
+	return imageKeyVals
+}
+
+func getFilePath(pathToBlueprint string) (string, error) {
+	blueprintURL, err := url.Parse(pathToBlueprint)
+	if blueprintURL == nil || err != nil {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is malformed. Error: %s", pathToBlueprint, err.Error())
+	}
+	if blueprintURL.Scheme == "" { // this is a local file
+		return pathToBlueprint, nil
+	}
+
+	// url
+	timeout := time.Duration(1 * time.Second)
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	tempDir, err := getTempDir()
+
+	if err != nil {
+		return "", fmt.Errorf("Local temp dir not properly defined. Error: %s", err.Error())
+	}
+	resp, err := client.Get(pathToBlueprint)
```

Comment:
Can we deal with redirections?

Reply:
Could you clarify further?

Reply:
I see we check status below and only consider `HTTP 200` responses. not 302 redirects, so not sure how this works, I just wanted to ask a control question, this may not even be relevant.

Reply:
Thanks fo rthe clarification afaik go http client will follow redirects: https://github.com/golang/go/issues/12705